views/js/app.js

```diff
@@ -16,48 +16,491 @@
  * @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
  * International Registered Trademark & Property of Blockonomics
  */
+service = angular.module("shoppingcart.services", ["ngResource"]);
 
-app = angular.module("blockonomics-invoice", ["monospaced.qrcode"]);
+var flyp_base = 'https://flyp.me/api/v1';
 
-app.config(function ($interpolateProvider) {
+service.factory('AltcoinNew', function($resource) {
+    var rsc = $resource(flyp_base + '/order/new');
+    return rsc;
+});
+
+service.factory('AltcoinAccept', function($resource) {
+    var rsc = $resource(flyp_base + '/order/accept');
+    return rsc;
+});
+
+service.factory('AltcoinCheck', function($resource) {
+    var rsc = $resource(flyp_base + '/order/check');
+    return rsc;
+});
+
+service.factory('AltcoinInfo', function($resource) {
+    var rsc = $resource(flyp_base + '/order/info');
+    return rsc;
+});
+
+service.factory('AltcoinAddRefund', function($resource) {
+    var rsc = $resource(flyp_base + '/order/addrefund');
+    return rsc;
+});
 
+service.factory('AltcoinLimits', function($resource) {
+    var rsc = $resource(flyp_base + '/order/limits/:coin/BTC', {
+        coin: '@coin'
+    });
+    return rsc;
+});
+
+service.factory('AltcoinAjax', function($resource) {
+    var rsc = $resource(ajax_url);
+    return rsc;
+});
+
+app = angular.module("blockonomics-invoice", ["monospaced.qrcode",  "shoppingcart.services"]);
+
+app.config(function($compileProvider) {
+    $compileProvider.aHrefSanitizationWhitelist(/^\s*(https?|ftp|mailto|data|chrome-extension|bitcoin|ethereum|litecoin):/);
+    // Angular before v1.2 uses $compileProvider.urlSanitizationWhitelist(...)
+});
+
+app.config(function ($interpolateProvider) {
     $interpolateProvider.startSymbol('//');
     $interpolateProvider.endSymbol('//');
 })
 
-app.controller("CheckoutController", function($window, $scope, $location, $interval, $rootScope) {
+function getParameterByNameBlocko(name, url) {
+    if (!url) {
+        url = window.location.href;
+    }
+    name = name.replace(/[\[\]]/g, "\\$&");
+    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+        results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+}
+
+app.controller("CheckoutController", function($window, $scope, $location, $interval, $rootScope, $httpParamSerializer, $timeout, AltcoinLimits, AltcoinNew, AltcoinAccept) {
     var totalProgress = 100;
     var totalTime = 10*60; //10m
     $scope.progress = totalProgress;
     $scope.clock = totalTime;
+    $scope.copyshow = false;
+
+    //Create url when the order is received 
+    $scope.finish_order_url = function() {
+        return final_url;
+    }
+
+    //Decode the tracking url
+    $scope.alt_decode_url = function(url) {
```

Comment:
Please use decodeURI or decodeURIComponent. if it doesn't work... add reference to stackexchange thread